nvflare/private/fed/server/server_command_agent.py

```diff
@@ -12,60 +12,105 @@
+    def start(self):
+        self.cell.register_request_cb(
+            channel=CellChannel.SERVER_COMMAND,
+            topic="*",
+            cb=self.execute_command,
         )
-        self.thread.start()
-        self.logger.info(f"ServerCommandAgent listening on port: {self.listen_port}")
-
-    def execute_command(self, conn, engine):
-        while not self.asked_to_stop:
-            try:
-                if conn.poll(1.0):
-                    msg = conn.recv()
-                    msg = fobs.loads(msg)
-                    command_name = msg.get(ServerCommandKey.COMMAND)
-                    data = msg.get(ServerCommandKey.DATA)
-                    command = ServerCommands.get_command(command_name)
-                    if command:
-                        with engine.new_context() as new_fl_ctx:
-                            reply = command.process(data=data, fl_ctx=new_fl_ctx)
-                            if reply is not None:
-                                conn.send(reply)
-            except EOFError:
-                self.logger.info("listener communication terminated.")
-                break
-            except Exception as e:
-                self.logger.error(
-                    f"IPC Communication error on the port: {self.listen_port}: {secure_format_exception(e)}."
-                )
+        self.cell.register_request_cb(
+            channel=CellChannel.AUX_COMMUNICATION,
+            topic="*",
+            cb=self.aux_communicate,
+        )
+        self.logger.info(f"ServerCommandAgent cell register_request_cb: {self.cell.get_fqcn()}")
+
+    def execute_command(self, request: CellMessage) -> CellMessage:
+
+        if not isinstance(request, CellMessage):
+            raise RuntimeError("request must be CellMessage but got {}".format(type(request)))
+
+        command_name = request.get_header(MessageHeaderKey.TOPIC)
+        data = fobs.loads(request.payload)
+
+        token = request.get_header(CellMessageHeaderKeys.TOKEN, None)
+        client = None
+        if token:
+            client = self._get_client(token)
+            if client:
+                data.set_header(ServerCommandKey.FL_CLIENT, client)
+        if command_name in ServerCommands.client_request_commands_names and not client:
+            return make_reply(ReturnCode.AUTHENTICATION_ERROR, "Request from invalid client", fobs.dumps(None))
+        command = ServerCommands.get_command(command_name)
+
+        if command:
+            with self.engine.new_context() as new_fl_ctx:
+                reply = command.process(data=data, fl_ctx=new_fl_ctx)
+                if reply is not None:
+                    return_message = new_cell_message({}, fobs.dumps(reply))
+                    return_message.set_header(MessageHeaderKey.RETURN_CODE, ReturnCode.OK)
+                else:
+                    return_message = make_reply(ReturnCode.PROCESS_EXCEPTION, "No process results", fobs.dumps(None))
+                return return_message
```

Comment:
if no command, what will be returned?
Should we have some responses that contains error information?